src/mo_frontend/typing.ml

```diff
@@ -2300,8 +2302,14 @@ and infer_block_exps env decs : T.typ =
 and infer_dec env dec : T.typ =
   let t =
   match dec.it with
-  | ExpD exp
-  | LetD (_, exp) ->
+  | ExpD exp 
+  | LetD (_, exp, None) -> infer_exp env exp
+  | LetD (_, exp, Some fail) -> 
+    let fail_type = infer_exp env fail in
+    if not T.(sub fail_type Non) then 
+      local_error env fail.at "M0500"
```

Comment:
I never know if this should be `error` or `local_error`...

Reply:
Also, pick a fresh code from `src/lang_utils/error_codes.ml`.

Reply:
Is there a scheme to these numbers? Picking 159 if not

Reply:
no, see the end of `src/lang_utils/error_codes.ml`

Reply:
Ah okay. Will fix